storage/remote/queue_manager_test.go

```diff
@@ -281,6 +281,49 @@ func TestReleaseNoninternedString(t *testing.T) {
 	testutil.Assert(t, metric == 0, "expected there to be no calls to release for strings that were not already interned: %d", int(metric))
 }
 
+func TestCalculateDesiredsShards(t *testing.T) {
+	type testcase struct {
+		startingShards int
+		reshard        bool
+	}
+	cases := []testcase{
+		{
+			// Test that ensures that if we haven't successfully sent a
+			// sample recently the queue will not reshard.
+			startingShards: 10,
+			reshard:        false,
+		},
+		{
+			startingShards: 5,
+			reshard:        true,
+		},
+	}
+	for _, c := range cases {
+		client := NewTestStorageClient()
+		m := NewQueueManager(nil, nil, "", newEWMARate(ewmaWeight, shardUpdateDuration), config.DefaultQueueConfig, nil, nil, client, defaultFlushDeadline)
+		m.numShards = c.startingShards
+		// Fake the samples in/out rates and the last sent timestamp so the reshard calculation has some numbers to use.
+		m.samplesIn.incr(1000)
```

Comment:
Consider including samplesIn and samplesOut in the testcases. Right now it would be hard to write new test cases with these hard coded values.